datasync/downloader/downloader.go

```diff
@@ -322,7 +322,11 @@ func (d *Downloader) Synchronise(id string, head common.Hash, td *big.Int, mode
 
 	if errors.Is(err, errInvalidChain) {
 		logger.Warn("Synchronisation failed, dropping peer", "peer", id, "err", err)
-		d.dropPeer(id)
+		if d.dropPeer == nil {
+			logger.Warn("Downloader wants to drop peer, but peerdrop-function is not set", "peer", id)
```

Comment:
How about setting dropPeer function when we create a new Downloader?

Reply:
The `dropPeer` is not nil on our codebase, but it is added for consistency with ethereum codebase, although this PR is not necessary. What do you think?

Reply:
I understood. Thanks!